src/foam/dao/FileJournal.js

```diff
@@ -237,9 +237,12 @@ foam.CLASS({
         }
       ],
       javaCode: `
-        BufferedWriter writer = getWriter();
+        java.io.Writer writer = getWriter();
         writer.write(data);
-        writer.newLine();
+        writer.write(System.getProperty("line.separator"));
+        // Should we flush() here?  It's probably faster not too as the writer
```

Comment:
I would argue on the "yes" side. If the thread (or program) gets killed before the buffer is written to file we don't end up with a corrupted record.